### ISSUE TITLE
Name "version::CLASS" used only once: possible typo at /opt/perl/lib/5.10.1/Safe.pm line 251.

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9827]
Migrated from [rt.perl.org#68530](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=68530) (status was 'resolved')

Searchable as RT68530$

**Comment by p5pRT** [comment]
### From lpsolit@gmail.com

This is a bug report for perl from LpSolit@&#8203;gmail\.com\,
generated with the help of perlbug 1\.39 running under perl 5\.10\.1\.




---
When I compile the test\.cgi script below\, the following warning is thrown&#8203;:

$ /opt/perl/bin/perl \-cwT test\.cgi
Name "version&#8203;::CLASS" used only once&#8203;: possible typo at
/opt/perl/lib/5\.10\.1/Safe\.pm line 251\.
test\.cgi syntax OK


test\.cgi&#8203;:

\#\!/opt/perl/bin/perl \-wT

use strict;
use lib '\.';
use Test5101;



Test5101\.pm&#8203;:

package Test5101;

use Safe;
my $s = new Safe;


This is the minimal testcase to reproduce the warning above\. If I copy
the content of Test5101\.pm into test\.cgi\, the warning goes away\. So this
means the error is only present if Safe\.pm is called from within the
Perl module\.

This error is only thrown from Perl 5\.10\.1 RC1\. No error is thrown with
Perl 5\.10\.0 and older\. As Perl 5\.10\.0 comes with Safe 2\.12 and Perl
5\.10\.1 with Safe 2\.17\, this may as well be a bug in Safe\.pm itself\.



---


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=medium
&nbsp;    module=Safe


---
Site configuration information for perl 5\.10\.1&#8203;:

Configured by buclin at Fri Aug 14 21&#8203;:44&#8203;:45 CEST 2009\.

Summary of my perl5 \(revision 5 version 10 subversion 1\) configuration&#8203;:

&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.6\.27\.24\-server\-1mnb\, archname=i686\-linux
&nbsp;    uname='linux localhost 2\.6\.27\.24\-server\-1mnb \#1 smp wed jun 10
17&#8203;:18&#8203;:14 edt 2009 i686 intel\(r\) core\(tm\)2 duo cpu e8400 @&#8203; 3\.00ghz gnulinux '
&nbsp;    config\_args='\-Dprefix=/opt/perl \-des'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=undef\, usemultiplicity=undef
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-strict\-aliasing \-pipe \-fstack\-protector
\-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O2'\,
&nbsp;    cppflags='\-fno\-strict\-aliasing \-pipe \-fstack\-protector
\-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.3\.2'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=4\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=/lib/libc\-2\.8\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.8'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-L/usr/local/lib
\-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    RC1

&nbsp;    PERL\_GIT\_UNPUSHED\_COMMITS        /\* do not remove this line \*/
&nbsp;    PERL\_GIT\_UNCOMMITTED\_CHANGES    /\* do not remove this line \*/



---
@&#8203;INC for perl 5\.10\.1&#8203;:
&nbsp;    /opt/perl/lib/5\.10\.1/i686\-linux
&nbsp;    /opt/perl/lib/5\.10\.1
&nbsp;    /opt/perl/lib/site\_perl/5\.10\.1/i686\-linux
&nbsp;    /opt/perl/lib/site\_perl/5\.10\.1
&nbsp;    \.



---
Environment for perl 5\.10\.1&#8203;:
&nbsp;    HOME=/root
&nbsp;    LANG=fr\_CH\.UTF\-8
&nbsp;    LANGUAGE=fr\_CH\.UTF\-8&#8203;:fr
&nbsp;    LC\_ADDRESS=fr\_CH\.UTF\-8
&nbsp;    LC\_COLLATE=fr\_CH\.UTF\-8
&nbsp;    LC\_CTYPE=fr\_CH\.UTF\-8
&nbsp;    LC\_IDENTIFICATION=fr\_CH\.UTF\-8
&nbsp;    LC\_MEASUREMENT=fr\_CH\.UTF\-8
&nbsp;    LC\_MESSAGES=fr\_CH\.UTF\-8
&nbsp;    LC\_MONETARY=fr\_CH\.UTF\-8
&nbsp;    LC\_NAME=fr\_CH\.UTF\-8
&nbsp;    LC\_NUMERIC=fr\_CH\.UTF\-8
&nbsp;    LC\_PAPER=fr\_CH\.UTF\-8
&nbsp;    LC\_SOURCED=1
&nbsp;    LC\_TELEPHONE=fr\_CH\.UTF\-8
&nbsp;    LC\_TIME=fr\_CH\.UTF\-8
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)

PATH=/sbin&#8203;:/usr/sbin&#8203;:/bin&#8203;:/usr/bin&#8203;:/usr/X11R6/bin&#8203;:/usr/local/bin&#8203;:/usr/local/sbin&#8203;:/root/bin
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @iabyn

On Fri\, Aug 14\, 2009 at 03&#8203;:05&#8203;:44PM \-0700\, Frédéric Buclin wrote&#8203;:
> \# New Ticket Created by  Frédéric Buclin 
> \# Please include the string&#8203;:  \[perl \#68530\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=68530 >
> 
> 
> This is a bug report for perl from LpSolit@&#8203;gmail\.com\,
> generated with the help of perlbug 1\.39 running under perl 5\.10\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> When I compile the test\.cgi script below\, the following warning is thrown&#8203;:
> 
> $ /opt/perl/bin/perl \-cwT test\.cgi
> Name "version&#8203;::CLASS" used only once&#8203;: possible typo at
> /opt/perl/lib/5\.10\.1/Safe\.pm line 251\.
> test\.cgi syntax OK
> 
> 
> test\.cgi&#8203;:
> 
> \#\!/opt/perl/bin/perl \-wT
> 
> use strict;
> use lib '\.';
> use Test5101;
> 
> 
> 
> Test5101\.pm&#8203;:
> 
> package Test5101;
> 
> use Safe;
> my $s = new Safe;
> 
> 
> This is the minimal testcase to reproduce the warning above\. If I copy
> the content of Test5101\.pm into test\.cgi\, the warning goes away\. So this
> means the error is only present if Safe\.pm is called from within the
> Perl module\.
> 
> This error is only thrown from Perl 5\.10\.1 RC1\. No error is thrown with
> Perl 5\.10\.0 and older\. As Perl 5\.10\.0 comes with Safe 2\.12 and Perl
> 5\.10\.1 with Safe 2\.17\, this may as well be a bug in Safe\.pm itself\.

Now fixed in blead with a930c511fcc6cf6f5ef849a9a4a028ff1cd6e27a
and fixed in 5\.10\.1\-to\-be with 9de9eeb1452212b331264652386099df5794d4a5\.

\-\- 
In England there is a special word which means the last sunshine
of the summer\. That word is "spring"\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
